samples/mqtt/basic_pub_sub/main.cpp

```diff
@@ -42,7 +46,19 @@ static void s_printHelp()
         "websockets)\n");
     fprintf(stdout, "proxy_host: if you want to use a proxy with websockets, specify the host here (optional).\n");
     fprintf(
-        stdout, "proxy_port: defaults to 8080 is proxy_host is set. Set this to any value you'd like (optional).\n\n");
+        stdout, "proxy_port: defaults to 8080 is proxy_host is set. Set this to any value you'd like (optional).\n");
+
+    fprintf(stdout, "  x509: Use the x509 credentials provider while using websockets (optional)\n");
+    fprintf(stdout, "  x509rolealias: Role alias to use with the x509 credentials provider (required for x509)\n");
+    fprintf(stdout, "  x509endpoint: Endpoint to fetch x509 credentials from (required for x509)\n");
+    fprintf(stdout, "  x509thing: Thing name to fetch x509 credentials on behalf of (required for x509)\n");
+    fprintf(
+        stdout,
+        "  x509cert: Path to the IoT thing certificate used in fetching x509 credentials (required for x509)\n");
+    fprintf(
+        stdout, "  x509key: Path to the IoT thing private key used in fetching x509 credentials (required for x509)\n");
+    fprintf(
+        stdout, "  x509rootca: Path to the root certificate used in fetching x509 credentials (required for x509)\n\n");
```

Comment:
I had a knee-jerk reaction that we should just re-use the existing `--key` `--cert` `--ca_file`.

But thinking about it, I agree that it does make clear that the existing flags are about direct connections and the new flags are for a different purpose

Reply:
We'd also need to add conditional skipping setting the base tls context, since if you try to make an http connection with key/cert based tls context it won't work.